ApplyToBecomeInternal/ApplyToBecomeInternal/Pages/ProjectList/Index.cshtml.cs

```diff
@@ -40,41 +39,25 @@ public IndexModel(IAcademyConversionProjectRepository repository)
 
 		public async Task OnGetAsync()
 		{
-			ApiResponse<ApiV2Wrapper<IEnumerable<AcademyConversionProject>>> response = await _repository.GetAllProjects(CurrentPage, _pageSize);
-			ApiResponse<List<string>> statusesResponse = await _repository.GetAvailableStatuses();
-			if (statusesResponse.Success) Filters.AvailableStatuses = statusesResponse.Body.ConvertAll(r => r.SentenceCase());
-
-			Projects = response.Body.Data.Select(Build).ToList();
-			HasNextPage = response.Body?.Paging?.NextPageUrl != null;
-			TotalProjects = response.Body?.Paging?.RecordCount ?? 0;
+			Filters.PopulateFrom(Request.Query);
 
-			Filters.AvailableDeliveryOfficers = Projects.Select(a => a.AssignedUserFullName).Where(s => !string.IsNullOrEmpty(s)).ToList();
-			if (CurrentPage - 5 > 1)
-			{
-				StartingPage = CurrentPage - 5;
-			}
-		}
-
-		public async Task<IActionResult> OnPostAsync()
-		{
-			string selectedStatuses = string.Join(',', Filters.SelectedStatuses);
-			ApiResponse<ApiV2Wrapper<IEnumerable<AcademyConversionProject>>> response =
-				await _repository.GetAllProjects(CurrentPage, _pageSize, Filters.SelectedOfficers.ToList(), Filters.Title, selectedStatuses);
-
-			ApiResponse<List<string>> availableStatuses = await _repository.GetAvailableStatuses();
-			if (availableStatuses.Success) Filters.AvailableStatuses = availableStatuses.Body;
+			ApiResponse<ApiV2Wrapper<IEnumerable<AcademyConversionProject>>> response = 
+				await _repository.GetAllProjects(CurrentPage, _pageSize, Filters.Title, Filters.SelectedOfficers, Filters.SelectedStatuses);
 
 			Projects = response.Body.Data.Select(Build).ToList();
 			HasNextPage = response.Body?.Paging?.NextPageUrl != null;
 			TotalProjects = response.Body?.Paging?.RecordCount ?? 0;
 
-			// Potentially create an endpoint for DO's in the future?
 			ApiResponse<ApiV2Wrapper<IEnumerable<AcademyConversionProject>>> allResponses = await _repository.GetAllProjects(CurrentPage, _pageSize);
-			Filters.AvailableDeliveryOfficers = allResponses.Body.Data.Select(Build).ToList().Select(a => a.AssignedUserFullName).Where(s => !string.IsNullOrEmpty(s)).ToList();
+			Filters.AvailableDeliveryOfficers = allResponses.Body.Data.Select(x => x.AssignedUser.FullName).Where(s => string.IsNullOrEmpty(s) is false).Distinct().ToList();
+
+			ApiResponse<List<string>> statusesResponse = await _repository.GetAvailableStatuses();
+			if (statusesResponse.Success) Filters.AvailableStatuses = statusesResponse.Body.ConvertAll(r => r.SentenceCase());
+
 			if (CurrentPage - 5 > 1)
+			{
 				StartingPage = CurrentPage - 5;
-
-			return Page();
```

Comment:
This is interesting, are we not required to return a Page() in the OnGet?

Reply:
Nope; the default 'result' is to render the associated template.